### PR TITLE
Securing Installation from Sql injection

### PR DESCRIPTION
**Pull request by MathurinNkenfack**
1.preventing sql injection with prepared statements
2. change utf8 to utf8mb4

### CODE REVIEW

**Comment by muarachmann** [comment]
Give this a proper title ...

File: library/classes/Installer.class.php [hunk 1]
```diff
@@ -473,304 +483,329 @@ public function write_configuration_file() {
...
+            $results = $this->dbh->query($sql);
+        }else{
+            $stmt = $this->dbh->prepare($sql);
+            if($count == 1){
+                $stmt->bind_param($str,$value[0]);
+            }
+            if($count == 2){
+                $stmt->bind_param($str,$value[0],$value[1]);
+            }
+            if($count == 3){
```
Comment (muarachmann) [inline library/classes/Installer.class.php]: No this is baaaaaad. Try another approach how about escaping the query then binding the array in here. Try another alternative.

Comment (muarachmann) [reply library/classes/Installer.class.php]: What if they were 20 params

Comment (MathurinNkenfack) [reply library/classes/Installer.class.php]: had issues with knowing how many elements the array has to index it and as I saw the number number coming from the user did not exceed 5...

Comment (MathurinNkenfack) [reply library/classes/Installer.class.php]: number of inputs*

Comment (MathurinNkenfack) [reply library/classes/Installer.class.php]: php > 5 doesn't support variable decomposition also

Comment (MathurinNkenfack) [reply library/classes/Installer.class.php]: if you have any clue...I will gladly like to hear

**Comment by muarachmann** [comment]
We are not using php5 and we wont offer support for that. A for loop can
do. Try to research more on that and ideally, you should be
escaping the parameters too. Something like
```$this->mysql_escape($this->login)```.

BR,
Mua

File: library/classes/Installer.class.php (hunk 1 above)
Comment (robbyoconnor) [reply library/classes/Installer.class.php]: This is where you communicate and ask questions.

Comment (MathurinNkenfack) [reply library/classes/Installer.class.php]: @robbyoconnor  thanks

**Comment by MathurinNkenfack** [comment]
@muarachmann please can u review the changes please

**Pull request closed without merging**